cabal-testsuite/PackageTests/Regression/T6011/pre-proc/pre-proc.cabal

```diff
@@ -0,0 +1,12 @@
+name:                pre-proc
+version:             999.999.999
+synopsis:            Checks build-tool-depends are put in PATH
+license:             BSD3
+category:            Testing
+build-type:          Simple
```

Comment:
This is not `Custom`, so it's not what described in #6011

I'll check what happens with non-Custom builds, there we should be able to build just an executable (of possibly multiple in the package). I don't defaulting to build every exe, I'd expect someone to complain about that too.

Reply:
Correct. It's closely related but not a direct reproduction. It _appears_ to be the case that custom builds are not the core issue, but the fact that custom turns off per-component. Whether or not it effects custom builds, this is an issue.